sentry-spring-boot-jakarta/src/main/java/io/sentry/spring/boot/jakarta/SentrySpanRestClientCustomizer.java

```diff
@@ -0,0 +1,28 @@
+package io.sentry.spring.boot.jakarta;
+
+import com.jakewharton.nopen.annotation.Open;
+import io.sentry.IHub;
+import io.sentry.spring.jakarta.tracing.SentrySpanClientHttpRequestInterceptor;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.boot.web.client.RestClientCustomizer;
+import org.springframework.web.client.RestClient;
+
+@Open
+class SentrySpanRestClientCustomizer implements RestClientCustomizer {
+  private final @NotNull SentrySpanClientHttpRequestInterceptor interceptor;
+
+  public SentrySpanRestClientCustomizer(final @NotNull IHub hub) {
+    this.interceptor = new SentrySpanClientHttpRequestInterceptor(hub);
+  }
+
+  @Override
+  public void customize(final @NotNull RestClient.Builder restClientBuilder) {
+    restClientBuilder.requestInterceptors(clientHttpRequestInterceptors -> {
```

Comment:
This is a tricky one, I saw the same concept in the `SentrySpanRestTemplateCustomizer` but I'm not sure if the contains can catch that a `SentrySpanClientHttpRequestInterceptor` is already registered or not, as it is being instantiated in the constructor, so different instances can be registered easily. We might need a proper equals in the `SentrySpanClientHttpRequestInterceptor`